nemo_aligner/data/nlp/builders.py

```diff
@@ -486,6 +503,17 @@ def build_dataloader(
         "pad_samples_to_global_batch_size": pad_samples_to_global_batch_size,
     }
 
+    if (
+        limit_train_batches is None
+        or (isinstance(limit_train_batches, float) and limit_train_batches > 1.0)
```

Comment:
I think a float value > 1 should trigger a crash rather than be truncated to 1, because it's most likely a user error (e.g. trying to set an integer N but accidentally making it a float)